src/widgets/SearchBar.js

```diff
@@ -37,30 +40,49 @@ export const SearchBar = ({
   debounceTimeout,
   ...textInputProps
 }) => {
+  const [textValue, setTextValue] = useState('');
+
+  // Set the internal text value if it differs from the passed
+  // value. Keeping the search bar text input up to date with
+  // user input while debouncing actual filtering of data.
+  useEffect(() => {
+    if (value !== textValue) setTextValue(value);
+  }, [value]);
+
   const internalViewStyle = { ...viewStyle, borderColor: color };
   const internalTextStyle = { ...textInputStyle, color };
 
-  const debouncedCallback = useMemo(
-    () => newValue => debounce(onChangeText(newValue), debounceTimeout),
-    [onChangeText]
+  // For Debouncing to funtion, need to use the same function as the
```

Comment:
`To debounce the function...`?

Reply:
3a42e4d98fc377162c598fa5d0a806304d370cb4